packages/flutter/test/material/app_bar_test.dart

```diff
@@ -2563,6 +2563,217 @@ void main() {
+                // single viewport.
+                return notification.depth <= 1;
+              },
+            ),
+            body: Row(
+              children: <Widget>[
+                Expanded(
+                  child: ListView(
+                    key: list1Key,
+                    children: <Widget>[
+                      Container(height: 1200, color: Colors.teal),
+                    ],
+                  ),
+                ),
+                Expanded(
+                  child: ListView(
+                    key: list2Key,
+                    children: <Widget>[
+                      Container(height: 1200, color: Colors.teal),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+        )
+      ));
+
+      // Initial state: AppBar should have the default color.
+      expect(getAppBarBackgroundColor(tester), defaultColor);
+
+      // Scroll the first list view.
+      await tester.drag(find.byKey(list1Key), const Offset(0, -300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should now have the scrolled color.
+      expect(getAppBarBackgroundColor(tester), scrolledColor);
+
+      // Scroll back to the top.
+      await tester.drag(find.byKey(list1Key), const Offset(0, 300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should be back to the default color.
+      expect(getAppBarBackgroundColor(tester), defaultColor);
+
+      // Scroll the second list view.
+      await tester.drag(find.byKey(list2Key), const Offset(0, -300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should again have the scrolled color.
+      expect(getAppBarBackgroundColor(tester), scrolledColor);
+
+      // Scroll back to the top.
+      await tester.drag(find.byKey(list2Key), const Offset(0, 300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should be back to the default color.
+      expect(getAppBarBackgroundColor(tester), defaultColor);
+    });
+
+    testWidgets('Material3 - scrolledUnderElevation should be applied when scrolled under with multiple list views with list scrolled', (WidgetTester tester) async {
+      final GlobalKey list1Key = GlobalKey();
+      final GlobalKey list2Key = GlobalKey();
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            appBar: AppBar(
+              elevation: 0,
+              backgroundColor: MaterialStateColor.resolveWith((Set<MaterialState> states) {
+                return states.contains(MaterialState.scrolledUnder)
+                  ? scrolledColor
+                  : defaultColor;
+              }),
+              title: const Text('AppBar'),
+              notificationPredicate: (ScrollNotification notification) {
+                // Represents both scroll views below being treated as a
+                // single viewport.
+                return notification.depth <= 1;
```

Comment:
And here. The depth would not be increased in this case. Each should have a depth of 0.